driver-core/src/main/com/mongodb/internal/TimeoutContext.java

```diff
@@ -200,30 +181,43 @@ public long getMaxAwaitTimeMS() {
         return timeoutSettings.getMaxAwaitTimeMS();
     }
 
-    public long getMaxTimeMS() {
-        return notNull("Should never be null", maxTimeSupplier.get());
-    }
+    public void runMaxTimeMSTimeout(final Runnable onInfinite, final LongConsumer onRemaining,
+            final Runnable onExpired) {
```

Comment:
Considered and tried various alternatives:

1. Returning the remaining time leaks internals (which has led to bugs).
2. Returning a live Timeout causes unified tests to fail, because a timeout of 100ms will be sent to the server as 99ms as soon as any nanoseconds have elapsed.
3. Could return a fixed value under a new type of timeout, which would pass that fixed value to its run method (that is, it would not be a timeout, but a configuration value provider); but this is a stretch, can't be compared for "earlier" against other timeouts. 

This chosen approach follows the pattern of existing timeout runs, and without leaking internals, breaking tests, or abusing the Timeout type.